lib/matplotlib/scale.py

```diff
@@ -458,6 +458,115 @@ def get_transform(self):
         return self._transform
 
 
+class AsinhScale(ScaleBase):
+    """
+    A quasi-logarithmic scale based on the inverse hyperbolic sine (asinh)
+
+    For values close to zero, this is essentially a linear scale,
+    but for larger values (either positive or negative) it is asymptotically
+    logarithmic. The transition between these linear and logarithmic regimes
+    is smooth, and has no discontinutities in the function gradient
+    in contrast to the "symlog" scale.
+    """
+
+    name = 'asinh'
+
+    def __init__(self, axis, *, a0=1.0, **kwargs):
+        """
+        Parameters
+        ----------
+        a0 : float, default: 1
+            The scale parameter defining the extent of the quasi-linear region.
+        """
+        super().__init__(axis)
+        if a0 <= 0.0:
+            raise ValueError("Scale parameter 'a0' must be strictly positive")
+        self.a0 = a0
+
+    def get_transform(self):
+        return self.AsinhTransform(self.a0)
+
+    def set_default_locators_and_formatters(self, axis):
+        axis.set(major_locator=AsinhScale.AsinhLocator(self.a0),
+                 major_formatter='{x:.3g}')
+
+    class AsinhTransform(Transform):
+        input_dims = output_dims = 1
+
+        def __init__(self, a0):
+            super().__init__()
+            self.a0 = a0
+
+        def transform_non_affine(self, a):
+            return self.a0 * np.arcsinh(a / self.a0)
+
+        def inverted(self):
+            return AsinhScale.InvertedAsinhTransform(self.a0)
+
+    class InvertedAsinhTransform(Transform):
+        input_dims = output_dims = 1
+
+        def __init__(self, a0):
+            super().__init__()
+            self.a0 = a0
+
+        def transform_non_affine(self, a):
+            return self.a0 * np.sinh(a / self.a0)
+
+        def inverted(self):
+            return AsinhScale.AsinhTransform(self.a0)
+
+    class AsinhLocator(Locator):
+        """
+        An axis tick locator specialized for the arcsinh scale
+
+        This is very unlikely to have any use beyond the AsinhScale class.
+        """
+        def __init__(self, a0, apx_tick_count=12):
+            """
+            Parameters
+            ----------
+            a0 : float
+                The scale parameter defining the extent
+                of the quasi-linear region.
+            apx_tick_count : int, default: 12
+                The approximate number of major ticks that will fit
+                along the entire axis
+            """
+            super().__init__()
+            self.a0 = a0
+            self.apx_tick_count = apx_tick_count
+
+        def __call__(self):
+            dmin, dmax = self.axis.get_data_interval()
+            return self.tick_values(dmin, dmax)
+
+        def tick_values(self, vmin, vmax):
+            # Construct a set of "on-screen" locations
```

Comment:
This isn't really "onscreen" this is the "axes" co-ordinate system.  If you want to be truly "onscreen" you _could_ adopt a heuristic similar to that used by MaxNLocator that takes into account the axes size and the fontsize.  Its not perfect, but for small subplots 12 ticks will be _far_ too many.